src/main/java/org/opentripplanner/transit/service/StopModel.java

```diff
@@ -57,14 +65,62 @@ public StopModel(StopModelBuilder builder) {
     reindex();
   }
 
+  /**
+   * Merge child into main. The child model must be created using the {@code main.withContext()}
+   * method, if not this method will fail! If a duplicate key exist, then child value is kept -
+   * this feature is normally not allowed, but not enforced here.
+   */
+  private StopModel(StopModel main, StopModel child) {
+    if (main.hashCode() != child.parentHash) {
+      throw new IllegalArgumentException(
+        "A Stop model can only be merged with its parent, this is done to avoid duplicates/gaps " +
+        "in the stopIndex."
+      );
+    }
+    this.parentHash = NO_PARENT;
+    this.stopIndexSize = child.stopIndexSize();
+    this.areaStopById = MapUtils.combine(main.areaStopById, child.areaStopById);
```

Comment:
I'm not sure if this is performance critical (I suspect it isn't) but `MapUtils.combine` is probably not as efficient as it could be. I wrote it with small maps in mind.

It probably doesn't matter though.

Reply:
I first wrote my own, then I changed to use yours - I do not think it matters that much - this is going to be refactored anyway, so for now I would just keep it. I can add a smal change to your method witch will make it faster for two Maps.

Reply:
I tested it, and it is called 6 times during the graph build for Entur, in total 45 ms on my machine - not something we care about.